packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -3513,25 +3535,58 @@ class RenderSemanticsGestureHandler extends RenderProxyBox {
   @override
   void describeSemanticsConfiguration(SemanticsConfiguration config) {
     super.describeSemanticsConfiguration(config);
-
+    final bool switchControlEnabled = RendererBinding.instance.window.accessibilityFeatures.switchControl;
+    final Map<CustomSemanticsAction, VoidCallback> overrides = <CustomSemanticsAction, VoidCallback>{};
     if (onTap != null && _isValidAction(SemanticsAction.tap))
       config.onTap = onTap;
     if (onLongPress != null && _isValidAction(SemanticsAction.longPress))
       config.onLongPress = onLongPress;
     if (onHorizontalDragUpdate != null) {
-      if (_isValidAction(SemanticsAction.scrollRight))
+      if (_isValidAction(SemanticsAction.scrollRight)) {
         config.onScrollRight = _performSemanticScrollRight;
-      if (_isValidAction(SemanticsAction.scrollLeft))
+        if (switchControlEnabled && _hasLabelOverride(SemanticsAction.scrollRight))
+          _insertSemanticsActionLabelOverride(SemanticsAction.scrollRight, overrides);
+      }
+      if (_isValidAction(SemanticsAction.scrollLeft)) {
         config.onScrollLeft = _performSemanticScrollLeft;
+        if (switchControlEnabled && _hasLabelOverride(SemanticsAction.scrollLeft))
+          _insertSemanticsActionLabelOverride(SemanticsAction.scrollLeft, overrides);
+      }
     }
     if (onVerticalDragUpdate != null) {
-      if (_isValidAction(SemanticsAction.scrollUp))
+      if (_isValidAction(SemanticsAction.scrollUp)) {
         config.onScrollUp = _performSemanticScrollUp;
-      if (_isValidAction(SemanticsAction.scrollDown))
+        if (switchControlEnabled && _hasLabelOverride(SemanticsAction.scrollUp))
+          _insertSemanticsActionLabelOverride(SemanticsAction.scrollUp, overrides);
+      }
+      if (_isValidAction(SemanticsAction.scrollDown)) {
         config.onScrollDown = _performSemanticScrollDown;
+        if (switchControlEnabled && _hasLabelOverride(SemanticsAction.scrollDown))
+          _insertSemanticsActionLabelOverride(SemanticsAction.scrollDown, overrides);
+      }
+    }
+    if (overrides.isNotEmpty) {
+      config.customSemanticsActions = overrides;
     }
   }
 
+  void _insertSemanticsActionLabelOverride(
+    SemanticsAction action,
+    Map<CustomSemanticsAction, VoidCallback> overrides
+  ) {
+    final CustomSemanticsAction override = CustomSemanticsAction.overridingActionLabel(
+      action: action,
+      label: _semanticsActionLabelOverrides[action],
+    );
+    // Overrides do not need a handler.
+    overrides[override] = null;
```

Comment:
We directly inject that in SemanticsNode.getSemanticsData() for hint override. If we want to do that we would have to make a general api in SemanticsNode, which i am a bit uncertain because this should really only been used for scrolling in switch control